hooks/pre-command

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euo pipefail
+
+echo "~~~ Installing nvm"
+
+NVM_DIR="$(mktemp -d)"
```

Comment:
Nitpick: how about providing a template for the name of the temp dir to make it less cryptic? (and easy to identify when browsing `/tmp` and/or if the `pre-exit` hook failed for some reason)

```suggestion
NVM_DIR="$(mktemp -d nvm-XXXXXXXXXX)"
```

Reply:
Good idea! I used https://github.com/Automattic/nvm-buildkite-plugin/pull/3/commits/6efbecf1db63a5d12d36a0b823630fa71ab169e5. The command you suggested creates a dir at the working directory, which probably is fine too. I used `-t` option to continue creating a dir at $TMPDIR, but with the given prefix.

Reply:
If I read the `man` page correctly, when using the `-t` option you should not provide the `XXX`s: the `-t` option will build the template for you, taking care of adding `$TMPDIR/` as prefix and the `.XXXXXXXX`s as suffix:

```sh
$ mktemp -dt automattic-nvm-XXXXXXXX
/var/folders/fn/g0dpb1lx345g5lg0s65wlp800000gn/T/automattic-nvm-XXXXXXXX.uYtXRNwr

$ mktemp -dt automattic-nvm
/var/folders/fn/g0dpb1lx345g5lg0s65wlp800000gn/T/automattic-nvm.7gRUjBbn
```

So I think we should use this instead:
```sh
NVM_DIR="$(mktemp -d -t automattic-nvm)"
```

Reply:
I [tried that](https://buildkite.com/automattic/nvm-buildkite-plugin/builds/21#018a29b7-76ce-4223-a58d-5027ac5293a1/158-159) and the command failed with error: "mktemp: too few X's in template ‘automattic-nvm’".

Apparently macOS mktemp and Linux mktemp takes `-t` differently. On macOS, the man page says it's a "prefix", where as on Linux `-t` is a "template".